### ISSUE TITLE
E1001 When running cfn-lint via AWS code build with .cfnlintrc.yaml

### ISSUE THREAD

**Issue by mtschneiders** [aws-cloudformation/cfn-lint#4147]
### CloudFormation Lint Version

1.35.2

### What operating system are you using?

Linux

### Describe the bug

We have a CodeBuild project that started failing after updating to version 1.35.2 of cfn-lint.
Note: we use .cfnlintrc.yaml to determine the templates that need to be linted.

buildspec.yml: 

```
version: 0.2
phases:
  build:
      commands:
        - cfn-lint --format junit 
```

Error after running cfn-lint:

```
[Container] 2025/05/21 19:43:05.707884 Phase context status code: COMMAND_EXECUTION_ERROR Message: Error while executing command: cfn-lint --format junit. Reason: exit status 2

E1001 Basic CloudFormation Template Configuration

Resources' is a required property
--
'Resources' is a required property at None:1:1
```

.cfnlintrc.yaml:

```
templates:
  - cicd/**/*.yml
  - templates/**/*.yml
ignore_templates:
  - cicd/buildspec.yml
output_file: ./reports/test/cfn-lint.xml
```

### Expected behavior

The command should not error out.

### Reproduction template

any template

**Comment by kddejong** [comment]
You mean `.cfnlintrc` file right?

**Comment by mtschneiders** [comment]
Yes, I meant `.cfnlintrc`, sorry.

**Comment by jakob-keller** [comment]
Same here with the following `.cfnlintrc.yaml`:

```yaml
include_checks:
  - I
regions:
  - eu-central-1
templates:
  - cloudformation/generic/*.template
  - cloudformation/global/*.template
  - cloudformation/instance-specific/*.template
```

**Comment by kddejong** [comment]
@jakob are you also using CodeBuild?  Or where is this running? Locally, CI/CD process?

**Comment by jakob-keller** [comment]
> [@jakob](https://github.com/jakob) are you also using CodeBuild? Or where is this running? Locally, CI/CD process?

It hit me locally. I use the pre-commit hook:

```yaml
  - repo: https://github.com/aws-cloudformation/cfn-python-lint
    rev: v1.35.2
    hooks:
      - id: cfn-lint-rc
```

**Comment by encee** [comment]
The issue is happening to us as well but within github actions, if we force downgrade the version it works again. 

I reproduced it locally by passing stdin to the process, it fails the same way.

```
# cfn-lint --version
cfn-lint 1.35.2

# echo | cfn-lint
E1001 'Resources' is a required property
None:1:1
```
force downgraded the version 

```
# cfn-lint --version
cfn-lint 1.35.1

# echo | cfn-lint                                      
W8001 Condition X not used
stack.yml:62:3
```

I think it has something to do with this change.... https://github.com/aws-cloudformation/cfn-lint/pull/4138/files possibly?

**Comment by jakob-keller** [comment]
Release 1.35.3 fixes this for me.
